### DISCUSSION TITLE
A GUI based (node?) editor for defining workflows and steps that will output an OpenJD template/bundle

### DISCUSSION THREAD

**Discussion by Ahuge** [OpenJobDescription/openjd-specifications#35] [Ideas]
Hi

Love the idea of OpenJD, I was thinking that some sort of visual editor to aid in the creation of OpenJD templates.

I could see some sort of "Qt Creator" analogous editor, potentially node based, that would allow for creation and validation of the resulting OpenJD Templates and bundles.

Part of this "Idea" is inviting other to determine if they would find this useful?
It could lower the barrier to entry for creating new workflows. Part of the goal for this editor would be creation and validation of the resulting templates. We all know it can be easy to make mistakes in creating yaml/json files.

I was thinking some sort of Qt application, with a node graph (Something like this? https://github.com/jchanvfx/NodeGraphQt) and editor panels to add properties.

Let me know if anyone would find this valuable.
-Alex

**Comment by ddneilson** [comment]
For my part, I think that this is something that would be amazing to see come out of the community. I've heard a few folk that love node-based UIs express interest in this sort of thing.

**Comment by pkpriority** [comment]
+1 love this idea, Alex! Especially if it can read a template to help visualize the existing dependent steps as node-networks too. It could be a part of the validation/troubleshooting workflow for people manually working with OpenJD.

**Comment by RandomInsano** [comment]
Just chiming in on the validation, there already exists a validation CLI so take a look at that repo here on GitHub. I think we had a VSCode integration idea at one point but I don’t recall if that happened. A node based editor for those workflows that need them would be really neat to see!

**Reply by pkpriority** [reply]
I was careful to say "part of the validation [...] workflow". The CLI validation checks syntax for sure but I like the idea of representing the steps in a node graph GUI for a tired human to glance-check if there are errors in the dependency sequencing.
